From 8ae4e33a4f41363c2622df33109c4e23f5507560 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 16 Aug 2011 16:17:43 +0800
Subject: [PATCH] wrhv/e500: stub emulation for TLB1CFG

We don't have the read permission to TLB1CFG register in GOS. So we
will get the following panic when reading it.

Kernel panic - not syncing: Unsupported read for spr 0x2b1 in
smp_store_cpu_info+0x40/0x64

Call Trace:
[cf81fea0] [c0007d38] show_stack+0x44/0x160 (unreliable)
[cf81fed0] [c05a2cec] panic+0x12c/0x1a4
[cf81ff20] [c00174dc] wrhv_mfspr+0x40/0xd4
[cf81ff30] [c05a6534] smp_store_cpu_info+0x40/0x64
[cf81ff50] [c06e5f88] smp_prepare_cpus+0x38/0x138
[cf81ff80] [c06e0264] kernel_init+0x70/0x278
[cf81fff0] [c0011d8c] original_kernel_thread+0x4c/0x68

Since we never use the contents of this register in GOS, add a stub
emulation to return 0 when we are reading it. The reason we choose to
return 0 here is that it is more easy to cause a misfunction if this
value is used unintentionally by some code in GOS.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/kernel/vbi/wrhv.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/vbi/wrhv.c b/arch/powerpc/kernel/vbi/wrhv.c
index e4670c0..725078c 100644
--- a/arch/powerpc/kernel/vbi/wrhv.c
+++ b/arch/powerpc/kernel/vbi/wrhv.c
@@ -1740,6 +1740,9 @@ unsigned int wrhv_mfspr(unsigned int sprn)
 			value = VB_STATUS_REGS_ACCESS(wr_status, svr);
 			break;
 
+		case SPRN_TLB1CFG:
+			break;
+
 		/*
 		 * We have the read/write permission to SPEFSCR in GOS,
 		 * so we should get its value directly.
-- 
1.7.0.4

